tests/backup_helper.go

```diff
@@ -10471,3 +10493,341 @@ func IsPxInstalled() bool {
 	numOfNodes := len(node.GetStorageDriverNodes())
 	return numOfNodes > 0
 }
+
+// ShareCluster shares a cluster with users and groups and optionally shares existing backups.
+func ShareCluster(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string, shareClusterBackups bool) (*api.ShareClusterResponse, error) {
+	backupDriver := Inst().Backup
+	shareClusterRequest := &api.ShareClusterRequest{
+		OrgId: BackupOrgID,
+		ClusterRef: &api.ObjectRef{
+			Name: clusterName,
+			Uid:  clusterUid,
+		},
+		// userid of the user(s) to share the cluster with
+		Users: users,
+		// group(s) to share the cluster with
+		Groups: groups,
+		// share_cluster_backups share is optional, if set to true, it will additionally share existing backups
+		ShareClusterBackups: shareClusterBackups,
+	}
+	resp, err := backupDriver.ShareCluster(ctx, shareClusterRequest)
+	if err != nil {
+		return resp, err
+	}
+	return resp, nil
+}
+
+// ShareClusterWithValidation shares a cluster with users and groups and optionally shares existing backups and validates the share.
+func ShareClusterWithValidation(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string, shareClusterBackups bool) (*api.ShareClusterResponse, error) {
```

Comment:
Can you re Use ShareCluster function and avoid duplicate code for Sharing the cluster.
10522 - 10540 is duplicate of function ShareCluster Please reuse

Reply:
Done